ui/apps/platform/src/Containers/Collections/CollectionForm.tsx

```diff
@@ -90,31 +95,80 @@ export type CollectionFormProps = {
     headerContent?: ReactElement;
 };
 
+function yupLabelRuleObject({ field }: ByLabelResourceSelector) {
```

Comment:
I'd be curious to see what a manual validation function with equivalent functionality looks like when plugged into Formik. My initial feeling here is that this is a ton of boilerplate that duplicates what we already get from TS, with a couple of small validation functions on the values.

Reply:
Indeed. Next best thing is `yup.ObjectSchema<Whatever>` used only in Policies so far:
* https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Policies/Wizard/policyValidationSchemas.ts
* https://github.com/jquense/yup#ensuring-a-schema-matches-an-existing-type